modules/core/shared/src/main/scala/net/BitVectorSocket.scala

```diff
@@ -35,31 +35,30 @@ object BitVectorSocket {
    * @param socket the underlying `Socket`
    * @group Constructors
    */
-  def fromSocket[F[_]](
-    socket:       Socket[F],
-    readTimeout:  Duration
-  )(
-    implicit ev: Temporal[F]
-  ): BitVectorSocket[F] =
+  def fromSocket[F[_]](socket: Socket[F], readTimeout: Duration, carryRef: Ref[F, Chunk[Byte]])(implicit F: Temporal[F]): BitVectorSocket[F] =
     new BitVectorSocket[F] {
-
-      val withTimeout: F[Chunk[Byte]] => F[Chunk[Byte]] = readTimeout match {
+      private val withTimeout: F[Option[Chunk[Byte]]] => F[Option[Chunk[Byte]]] = readTimeout match {
         case _: Duration.Infinite   => identity
         case finite: FiniteDuration => _.timeout(finite)
       }
```

Comment:
`.timeout(...)` now supports `Duration.Infinite` so this helper shouldn't be needed anymore.

Reply:
Yeah I tried that but it benchmarked a bit slower since the short circuit happens once with current implementation but on each call with the new CE support.

Reply:
I'm amazed that makes such a difference 😅